### PR TITLE
Add solution file

### LINKED ISSUE #4488: Cannot build solution locally
### Description

As part of #4483 I tried to build the solution locally, but package restore fails.

At first it failed with this error:

```
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.ManifestTool.CrossPlatform (>= 2.1.3)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.VisualStudioEng.MicroBuild.Core (>= 1.0.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.VisualStudioEng.MicroBuild.Plugins.SwixBuild (>= 1.1.286)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\extensions.sln]
```

Then I edited NuGet.config to add the following:

```
<packageSource key="dotnet-core-internal-tooling">
  <package pattern="*" />
</packageSource>
```

Now the build fails with the following error because I don't have access to the AzDo package feed for DevDiv:

```
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1301: Unable to load the service index for source https://pkgs.dev.azure.com/devdiv/_packaging/dotnet-core-internal-tooling/nuget/v3/index.json. [C:\Coding\dotnet\extensions\extensions.sln]
```

### Reproduction Steps

Run `restore.cmd` while not being a Microsoft employee.

### Expected behavior

The solution successfully builds.

### Actual behavior

The solution fails to restore NuGet packages.

### Regression?

Unsure.

### Known Workarounds

Get hired by Microsoft to work in DevDiv 😄

### Configuration

8.0.100-rc.2.23462.28

### Other information

_No response_

### PR DESCRIPTION
**Pull request by martincostello**
Add a "kitchen sink" solution file by default so that the C# Dev Kit doesn't try to automatically create a solution file for itself when opening the repository in Visual Studio Code, which then breaks the build scripts such as `restore.cmd`.

The `.sln` file was generated by running `.\build.cmd -vs *` and then renaming the result to `extensions.sln` so it doesn't conflict with existing local development workflows generating subsets as `SDK.sln`.

Resolves #4488.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4509)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -308,4 +309,4 @@ BenchmarkDotNet.artifacts/
```
Comment (danmoseley) [inline .gitignore:309]: do you want to update building.md where it says "this repository does not contain a solution file"

btw, are things useable if you open this? the building.md steers towards creating sub-solutions on the fly, but perhaps VS can actually work OK with the whole thing.

**Review by danmoseley — APPROVED** [review]
thanks

File: .gitignore (hunk 1 above)
Comment (martincostello) [reply .gitignore:309]: Sure I can look at updating that too.

I did open it in VS and build it to check it compiled.

**Comment by RussKie** [comment]
I've  already travelled this road twice, and twice I had to retreat... Third time lucky?
I'll defer this to @geeknoid and @joperezr .

**Comment by RussKie** [comment]
I pinged the devkit team to see if something can be done to disable the auto-generation behavour.

File: extensions.sln [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Extensions.EnumStrings.Tests", "test\Libraries\Microsoft.Extensions.EnumStrings.Tests\Microsoft.Extensions.EnumStrings.Tests.csproj", "{8EC4BF46-73DE-46E0-B509-50F09677B5D6}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Extensions.Hosting.Testing.Tests", "test\Libraries\Microsoft.Extensions.Hosting.Testing.Tests\Microsoft.Extensions.Hosting.Testing.Tests.csproj", "{F40AA953-1763-41B2-9484-EB1C68821F43}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Extensions.Http.AutoClient.Tests", "test\Libraries\Microsoft.Extensions.Http.AutoClient.Tests\Microsoft.Extensions.Http.AutoClient.Tests.csproj", "{2449EC1F-E034-4DD2-887B-B4C71DAA855C}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Extensions.Http.Diagnostics.Tests", "test\Libraries\Microsoft.Extensions.Http.Diagnositcs.Tests\Microsoft.Extensions.Http.Diagnostics.Tests.csproj", "{90A6A3FA-842E-4B0B-84F5-AB5644469287}"
```
Comment (danmoseley) [inline extensions.sln:164]:
```suggestion
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Extensions.Http.Diagnostics.Tests", "test\Libraries\Microsoft.Extensions.Http.Diagnostics.Tests\Microsoft.Extensions.Http.Diagnostics.Tests.csproj", "{90A6A3FA-842E-4B0B-84F5-AB5644469287}"
```

build break

**Comment by danmoseley** [comment]
```
[ERROR] Multiple .sln files found in the root of the repository. Use --projects to specify the one you wish to build.
```
agh

**Comment by geeknoid** [comment]
We specifically operate without a solution file in this repo, slngen is the model we use here. I don't see a need to change this right now.

Thanks.

**Comment by RussKie** [comment]
> We specifically operate without a solution file in this repo, slngen is the model we use here. I don't see a need to change this right now.

@geeknoid this approach doesn't appear to be working for VS Code DevKit, which may be an IDE of choice so some contributors.

. . .

I had a quick offline chat with @michael-eng, and he pointed me to the settings that controls the sln file creation
image

@martincostello could you see if disabling this solves your issue?

**Comment by martincostello** [comment]
#4517

**Pull request closed without merging**